packages/flutter/test/cupertino/switch_test.dart

```diff
@@ -0,0 +1,35 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/widgets.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+void main() {
+  testWidgets('Switch can toggle on tap', (WidgetTester tester) async {
+    Key switchKey = new UniqueKey();
+    bool value = false;
+    await tester.pumpWidget(
+      new StatefulBuilder(
+        builder: (BuildContext context, StateSetter setState) {
+          return new Center(
+            child: new CupertinoSwitch(
+              key: switchKey,
+              value: value,
+              onChanged: (bool newValue) {
+                setState(() {
+                  value = newValue;
+                });
+              },
+            ),
+          );
+        },
+      ),
+    );
+
+    expect(value, isFalse);
+    await tester.tap(find.byKey(switchKey));
+    expect(value, isTrue);
+  });
+}
```

Comment:
I see what you're saying (that this test is really shallow), but I'm not quite ready for that yet.  It's also likely we'll need to tweak the parameters for the switch a bunch before perfect...